src/test/miner_tests.cpp

```diff
@@ -26,6 +27,25 @@
 
 BOOST_FIXTURE_TEST_SUITE(miner_tests, TestingSetup)
 
+// Validation errors used in tests (prevent typos and facilitate refactoring):
+#define REJECT_INVALID_BAD_BLK_SIGOPS "bad-blk-sigops"
+#define REJECT_INVALID_BAD_CB_MULTIPLE "bad-cb-multiple"
+#define REJECT_INVALID_BAD_TXNS_INPUTS_MISSINGORSPENT "bad-txns-inputs-missingorspent"
+#define REJECT_INVALID_BLOCK_VALIDATION_FAILED "block-validation-failed"
+
+// BOOST_CHECK_EXCEPTION predicates to check the specific validation error
+class CheckRejectInvalid {
+public:
+    CheckRejectInvalid(std::string const& arg) {
```

Comment:
```cpp
CheckRejectInvalid(const std::string& reason) : m_reason(reason) {}
```

Reply:
Thanks. If it wasn't obvious, I'm not terribly familiar with C++. :-)

Reply:
Done, and renamed the private variable from `reason` to `m_reason`